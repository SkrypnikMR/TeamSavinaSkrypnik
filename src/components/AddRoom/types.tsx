import React from 'react';

export interface IDiv {
    background: string;
    bottom: string;
    right: string;
    padding: string;
    fontSize: string;
    fontWeight: string;
    width: string;
    height: string;
    color: string;
    borderRadius: string;
    lineHeight: string;
    textAlign: string;
    margin: string;
    cursor: string;
    webkitTransform: string;
    msTransform: string;
    transform: string;
    boxShadow: string;
    transitionDuration: string;
}
