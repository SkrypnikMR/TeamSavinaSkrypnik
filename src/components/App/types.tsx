import { colors } from '../UI/baseLayout';

export interface ITheme {
    theme: 'dark' | 'light';
    colors: typeof colors;
}

export interface IDiv extends ITheme {
    minHeight: string;
    background: string;
    backgroundSize: string;
    direction: string;
}
