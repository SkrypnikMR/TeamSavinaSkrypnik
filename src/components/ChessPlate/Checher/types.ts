export type TChecker = {
    color: string;
}
