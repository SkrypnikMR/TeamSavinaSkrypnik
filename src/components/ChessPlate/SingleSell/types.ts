export type TSingleSell = {
    id: number;
};
