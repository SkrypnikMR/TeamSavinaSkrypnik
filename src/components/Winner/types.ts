export type TWinner = {
    winner?: string;
    userLogin: string;
}
