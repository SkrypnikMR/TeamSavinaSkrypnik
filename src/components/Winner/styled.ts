import styled from 'styled-components';
import { white } from '../UI/baseLayout';

export const StWinnerWrapper = styled.div`
        display:flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        height: 300px;
        margin-top: 150px;
        color: ${white};
        background: ${({ colors, theme }) => colors[theme].background};

`;
