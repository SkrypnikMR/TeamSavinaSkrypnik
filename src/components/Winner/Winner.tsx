import React from 'react';
import { TWinner } from './types';
import { StWinnerWrapper } from './styled';

function Winner({ winner }: TWinner) {
    return (
        <StWinnerWrapper>
            {winner}
        </StWinnerWrapper>
    );
}

export default Winner;
