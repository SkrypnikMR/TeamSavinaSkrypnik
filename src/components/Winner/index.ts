import { connect } from 'react-redux';
import { getUserLogin } from '../../store/game/selectors';
import Winner from './Winner';

const mapStateToProps = store => ({ userLogin: getUserLogin(store) });

export default connect(mapStateToProps)(Winner);
