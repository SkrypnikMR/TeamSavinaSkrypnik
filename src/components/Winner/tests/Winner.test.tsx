import React from 'react';
import { shallow, mount } from 'enzyme';
import { TWinner } from '../types';
import Winner from '../Winner';

describe('Winner', () => {
    let props: TWinner;
    beforeEach(() => {
        props = {
            winner: 'Ya',
            userLogin: 'neYa',
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<Winner {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should render StWinnerWrapper', () => {
        props = {
            winner: 'Ya',
            userLogin: 'Ya',
        };
        const component = mount(<Winner {...props} />);
        expect(component.find('styled__StWinnerWrapper')).toHaveLength(1);
    });
});
