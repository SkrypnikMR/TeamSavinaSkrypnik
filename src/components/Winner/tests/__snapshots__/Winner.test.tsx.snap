// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner Should match snapshot 1`] = `"<div class=\\"styled__StWinnerWrapper-sc-nohrb0-0 jmYrjM\\">Ya</div>"`;
